lib/drives/pages/drive_chat_page.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flutter/material.dart';
+
+import '../../account/models/profile.dart';
+import '../../rides/models/ride.dart';
+import '../../util/trip/pending_ride_card.dart';
+import '../../util/supabase.dart';
+import '../models/drive.dart';
+
+class DriveChatPage extends StatefulWidget {
+  final Drive drive;
+  const DriveChatPage({required this.drive, super.key});
+
+  @override
+  State<DriveChatPage> createState() => _DriveChatPageState();
+}
+
+class _DriveChatPageState extends State<DriveChatPage> {
+  late Drive _drive;
+
+  @override
+  void initState() {
+    super.initState();
+    _drive = widget.drive;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Set<Profile> riders = _drive.approvedRides!.map((ride) => ride.rider!).toSet();
+    List<Ride> pendingRides = _drive.pendingRides!.toList();
+    List<Widget> widgets = [];
+    if (riders.isEmpty && pendingRides.isEmpty) {
+      widgets.add(const Center(
+        child: Text("No riders or pending rides"),
+      ));
+    } else {
+      if (riders.isNotEmpty) {
+        List<Widget> riderColumn = [
+          Text(
+            "Riders",
```

Comment:
Please add some vertical space above this

Reply:
i did that. Thanks for the hint.